python/social_bot/tasks.py

```diff
@@ -812,66 +812,69 @@ def task_specific_observation(self, agent):
+        Agent will receive 100 when succefully kick the ball into the gate.
         Agent will receive the speed of getting closer to the ball before touching the
             ball within 45 degrees of agent direction. The reward is trunked within
             parameter target_speed.
-        Agent will receive negative normalized distance from ball to goal after
-            touching the ball within the direction. An offset of "target_speed + 1" is
-            included since touching the goal must be better than not touching.
+        Agent will receive negative normalized distance from ball to gate center
+            after touching the ball within the direction. An offset of
+            "target_speed + 1" is included since touching the ball must be better
+            than not touching.
 
     If no reward shaping, then the agent will only get -1/0/1 rewards.
     """
 
     def __init__(self,
                  env,
                  max_steps,
-                 goal_name="goal",
-                 success_distance_thresh=0.5,
+                 gate_distance=5.0,
                  random_range=4.0,
                  target_speed=2.0,
                  reward_weight=1.0,
                  sparse_reward=False):
         """
         Args:
             env (gym.Env): an instance of Environment
-            max_steps (int): episode will end if not reaching goal in so many steps
-            goal_name (string): name of the goal in the world
-            success_distance_thresh (float): the goal is reached if it's within this distance to the agent
-            random_range (float): the goal's random position range
+            max_steps (int): episode will end if the task is not achieved in so
+                many steps
+            gate_distance (float): the distance from the gate to the ball on
+                average. A smaller distance makes the kicking task easier.
+            random_range (float): the ball's random position range
             target_speed (float): the target speed runing to the ball. The agent will receive no more
                 higher reward when its speed is higher than target_speed.
             reward_weight (float): the weight of the reward
             sparse_reward (bool): if True, the agent will only get -1/0/1 rewards.
         """
         super().__init__(
             env=env, max_steps=max_steps, reward_weight=reward_weight)
-        self._goal_name = goal_name
         self._random_range = random_range
-        self._success_distance_thresh = success_distance_thresh
         self._target_speed = target_speed
         self._sparse_reward = sparse_reward
+        self._gate_distance = gate_distance
+        # By looking up the 'robocup_3Dsim_goal' model file:
+        self._gate_width = 2.1
+        self._gate_post_radius = 0.05
         self._env.insert_model(
             model="robocup_3Dsim_goal",
-            name="goal",
-            pose="-5.0 0 0 0 -0 3.14159265")
+            name="gate",
```

Comment:
Actually, I believe goal is the correct way to name it.

Reply:
I can change it back. Just thought that 'goal' is easily confused with the general 'goal' meaning in RL.

Reply:
> Actually, I believe goal is the correct way to name it.

I can change it back. Just thought that 'goal' is easily confused with the general 'goal' meaning in RL.